tree/ntuple/v7/inc/ROOT/RField.hxx

```diff
@@ -153,6 +158,12 @@ protected:
    /// Returns an index that can be used to remove the callback.
    size_t AddReadCallback(ReadCallback_t func);
    void RemoveReadCallback(size_t idx);
+   /// Register post-read callbacks corresponding to a list of ROOT I/O customization rules. `classp` is used to
+   /// fill the `TVirtualObject` instance passed to the user function.
+   void AddReadCallbacksFromIORules(const std::span<const TSchemaRule *> rules, TClass *classp = nullptr);
+   /// Called by `ConnectPageSource()` to register read callbacks only once connected; derived classes may override this
+   /// as appropriate
+   virtual void RegisterReadCallbacks() {}
```

Comment:
The problem is that we register (or not) the corresponding function for each read rule, depending on the on-disk C++ type version, which is only available after the field has been connected to a page source.

Therefore, we need something, e.g. such virtual member function, to notify derived classes that this information is available now.  I named it `RegisterReadCallbacks()` but it might perfectly be `OnConnectPageSource()` and let derived classes do further processing there, if required.

Reply:
Oh, right!  I think we should go then for `OnConnectPageSource()`